### PR TITLE
chore: fix PR Labeler config 

### PR DESCRIPTION
**Pull request by DanStough**
### Description
Fixing the format of the PR Labeler config file since we bumped the action to V5.

### CODE REVIEW

**Review by zalimeni — APPROVED** [review]

File: .github/pr-labeler.yml [hunk 1]
```diff
@@ -2,65 +2,107 @@
 # SPDX-License-Identifier: BUSL-1.1
 
 pr/dependencies:
-    - vendor/**/*
-    - go.*
+  - changed-files:
+    - any-glob-to-any-file:
+      - vendor/**/*
```
Comment (nathancoleman) [inline .github/pr-labeler.yml:7]: I haven't seen a reference to a `vendor` directory in sooo long 🤣

**Review by nathancoleman — APPROVED** [review]

**Comment by DanStough** [comment]
Passing now: https://github.com/hashicorp/consul/actions/runs/9164671273/job/25196439830?pr=21143

Adding the backport label.

**Comment by hc-github-team-consul-core** [comment]
@DanStough, a backport is missing for this PR [21141] for versions [1.18] please perform the backport manually and add the following snippet to your backport PR description:
```
<details>
	<summary> Overview of commits </summary>
		- <<backport commit 1>>
		- <<backport commit 2>>
		...
</details>
```

**Comment by hc-github-team-consul-core** [comment]
@DanStough, a backport is missing for this PR [21141] for versions [1.16,1.17] please perform the backport manually and add the following snippet to your backport PR description:
```
<details>
	<summary> Overview of commits </summary>
		- <<backport commit 1>>
		- <<backport commit 2>>
		...
</details>
```

**Pull request merged**